testsuite/features/reposync/reference_srv_check_reposync.feature

```diff
@@ -23,7 +23,7 @@ Feature: Reposync works as expected
 
 @uyuni
   Scenario: Check reposync of Ubuntu 22.04 channels being finished
-    Then I wait until the channel "ubuntu-22.04-pool-amd64-uyuni" has been synced
+    Then I wait until the channel "ubuntu-2204-pool-amd64-uyuni" has been synced
```

Comment:
nor that one...

Reply:
No possible channel exist with `.` in ubuntu name (checked on uyuni:master)
```
wiktor-srv:~ # spacewalk-common-channels --list | grep ubuntu-22
 ubuntu-2204-amd64-main-backports-uyuni: amd64-deb
 ubuntu-2204-amd64-main-security-uyuni: amd64-deb
 ubuntu-2204-amd64-main-updates-uyuni: amd64-deb
 ubuntu-2204-amd64-main-uyuni: amd64-deb
 ubuntu-2204-amd64-multiverse-backports-uyuni: amd64-deb
 ubuntu-2204-amd64-multiverse-security-uyuni: amd64-deb
 ubuntu-2204-amd64-multiverse-updates-uyuni: amd64-deb
 ubuntu-2204-amd64-multiverse-uyuni: amd64-deb
 ubuntu-2204-amd64-restricted-backports-uyuni: amd64-deb
 ubuntu-2204-amd64-restricted-security-uyuni: amd64-deb
 ubuntu-2204-amd64-restricted-updates-uyuni: amd64-deb
 ubuntu-2204-amd64-restricted-uyuni: amd64-deb
 ubuntu-2204-amd64-universe-backports-uyuni: amd64-deb
 ubuntu-2204-amd64-universe-security-uyuni: amd64-deb
 ubuntu-2204-amd64-universe-updates-uyuni: amd64-deb
 ubuntu-2204-amd64-universe-uyuni: amd64-deb
 ubuntu-2204-amd64-uyuni-client: amd64-deb
 ubuntu-2204-amd64-uyuni-client-devel: amd64-deb
 ubuntu-2204-pool-amd64-uyuni: amd64-deb
```

Reply:
same additional test, please

Reply:
After running the scenario:

```
  Scenario: Add Ubuntu 22.04                                                                    # features/build_validation/reposync/srv_sync_all_products.feature:658
      This scenario ran at: 2024-02-29 12:17:52 +0000
    When I use spacewalk-common-channel to add all "ubuntu-2204" channels with arch "amd64-deb" # features/step_definitions/command_steps.rb:182
      Channel ubuntu-2204-pool-amd64-uyuni added
      Channel ubuntu-2204-amd64-main-security-uyuni added
      Channel ubuntu-2204-amd64-main-updates-uyuni added
      Channel ubuntu-2204-amd64-main-uyuni added
      Channel ubuntu-2204-amd64-universe-backports-uyuni added
      Channel ubuntu-2204-amd64-universe-security-uyuni added
      Channel ubuntu-2204-amd64-universe-updates-uyuni added
      Channel ubuntu-2204-amd64-universe-uyuni added
      Channel ubuntu-2204-amd64-uyuni-client-devel added
...
```

Here is an interesting topic, despite I have no mentiones of `ubuntu-22.04` phrase in the whole repository, it still tries to synchronize it:

File is named `wiktor-srv:/var/log/rhn/reposync/` and here is it's content:

```
2024/02/29 13:50:54 +02:00 Command: ['/usr/bin/spacewalk-repo-sync', '--channel', 'ubuntu-22.04-pool-amd64-uyuni', '--type', 'deb', '--non-interactive']
2024/02/29 13:50:54 +02:00 Sync of channel started.
2024/02/29 13:50:54 +02:00 Repo URL: http://localhost/pub/repositories/empty-deb/?uniquekey=2204-uyuni
2024/02/29 13:50:54 +02:00     Packages in repo:                 0
2024/02/29 13:50:54 +02:00     No new packages to sync.
2024/02/29 13:50:54 +02:00 Filtering packages that failed to download
2024/02/29 13:50:54 +02:00 Importing packages started.
2024/02/29 13:50:54 +02:00 
2024/02/29 13:50:54 +02:00   Importing packages to DB:
2024/02/29 13:50:54 +02:00 Importing packages finished.
2024/02/29 13:50:54 +02:00 
2024/02/29 13:50:54 +02:00   Patches in repo: 0.
2024/02/29 13:50:54 +02:00 Sync completed.
```

and after manual verification of this `Command` on the version without `.` it does not recognize it:
```
wiktor-srv:/var/log/rhn/reposync # /usr/bin/spacewalk-repo-sync --channel ubuntu-2204-pool-amd64-uyuni --type deb --non-interactive
Channel ubuntu-2204-pool-amd64-uyuni is not custom or does not exist.
```

Investigating further it's the only channel with `localhost` as a repository:
```
wiktor-srv:/var/log/rhn/reposync # spacewalk-repo-sync --list | grep ubu
15:00:58 ubuntu-2204-amd64-universe-backports-uyuni | http://archive.ubuntu.com/ubuntu/dists/jammy-backports/universe/binary-amd64/
15:00:58 ubuntu-2204-amd64-universe-updates-uyuni | http://archive.ubuntu.com/ubuntu/dists/jammy-updates/universe/binary-amd64/
15:00:58 ubuntu-2204-amd64-main-uyuni | http://archive.ubuntu.com/ubuntu/dists/jammy/main/binary-amd64/
15:00:58 ubuntu-2204-amd64-main-updates-uyuni | http://archive.ubuntu.com/ubuntu/dists/jammy-updates/main/binary-amd64/
15:00:58 ubuntu-22.04-pool-amd64-uyuni | http://localhost/pub/repositories/empty-deb/?uniquekey=2204-uyuni
15:00:58 ubuntu-2204-amd64-universe-uyuni | http://archive.ubuntu.com/ubuntu/dists/jammy/universe/binary-amd64/
15:00:58 ubuntu-2204-amd64-universe-security-uyuni | http://archive.ubuntu.com/ubuntu/dists/jammy-security/universe/binary-amd64/
15:00:58 ubuntu-2204-amd64-uyuni-client-devel | https://download.opensuse.org/repositories/systemsmanagement:/Uyuni:/Master:/Ubuntu2204-Uyuni-Client-Tools/xUbuntu_22.04/
15:00:58 ubuntu-2204-amd64-main-security-uyuni | http://archive.ubuntu.com/ubuntu/dists/jammy-security/main/binary-amd64/
```

which seems to have no packages available:
```
wiktor-srv:/usr/share/susemanager/www/pub/repositories/empty-deb # cat Release 
Architectures: all
MD5Sum:
  <X> 0 Packages
SHA1:
  <X> 0 Packages
SHA256:
  <X> 0 Packages
```

`gobuster`ing http://archive.ubuntu.com/ubuntu/dists there is no such a catalog having `*pool`.
Maybe this channel should be uterly removed?

Reply:
1) here we see a difference between 22.04 and 2204 :crying_cat_face: 

2) I know @srbarrios  has some code for repos without packages, so that's definitely something that happens. But maybe there's no connection to what you see here.

About this problem with `localhost`, yes, it could be a bug. See with developers, we could for example have an error in the `.ini` file taken as input in `spacewalk-common-channels`.

In any case I don't see these specific problems linked to your PR.

But the problem of the dot in 22.04 is linked to your PR, yes.